src/Uno.UI/UI/Xaml/Controls/Repeater/ItemsRepeater.cs

```diff
@@ -0,0 +1,821 @@
+		Point m_layoutOrigin;
+
+		// Loaded events fire on the first tick after an element is put into the tree 
+		// while unloaded is posted on the UI tree and may be processed out of sync with subsequent loaded
+		// events. We keep these counters to detect out-of-sync unloaded events and take action to rectify.
+		int _loadedCounter;
+		int _unloadedCounter;
+
+		// Bug in framework's reference tracking causes crash during
+		// UIAffinityQueue cleanup. To avoid that bug, take a strong ref
+		IElementFactory m_itemTemplate;
+		Layout m_layout;
+		ElementAnimator m_animator;
+
+		// Bug where DataTemplate with no content causes a crash.
+		// See: https://github.com/microsoft/microsoft-ui-xaml/issues/776
+		// Solution: Have flag that is only true when DataTemplate exists but it is empty.
+		bool m_isItemTemplateEmpty;
+
+
+		public ItemsRepeater()
+		{
+			//__RP_Marker_ClassById(RuntimeProfiler.ProfId_ItemsRepeater);
+
+			_repeaterChildren = new UIElementCollection(this);
+			m_animationManager = new AnimationManager(this);
+			m_viewManager = new ViewManager(this);
+			//if (SharedHelpers.IsRS5OrHigher())
+			{
+				m_viewportManager = new ViewportManagerWithPlatformFeatures(this);
+			}
+			//else
+			//{
+			//	m_viewportManager = std.new ViewportManagerDownLevel(this);
+			//}
+
+			AutomationProperties.SetAccessibilityView(this, AccessibilityView.Raw);
+			//if (SharedHelpers.IsRS3OrHigher())
+			{
+				TabFocusNavigation = KeyboardNavigationMode.Once;
+				XYFocusKeyboardNavigation = XYFocusKeyboardNavigationMode.Enabled;
+			}
+
+			Loaded += OnLoaded;
+			Unloaded += OnUnloaded;
+
+			// Initialize the cached layout to the default value
+			var layout = Layout as VirtualizingLayout;
+			OnLayoutChanged(null, layout);
+		}
+
+		~ItemsRepeater()
+		{
+			m_itemTemplate = null;
+			m_animator = null;
+			m_layout = null;
+		}
+
+		#region IUIElementOverrides
+
+		protected override AutomationPeer OnCreateAutomationPeer()
+		{
+			return new RepeaterAutomationPeer(this);
+		}
+
+		#endregion
+
+		#region IUIElementOverrides7
+
+		protected override IEnumerable<DependencyObject> GetChildrenInTabFocusOrder()
+		{
+			return CreateChildrenInTabFocusOrderIterable();
+		}
+
+		#endregion
+
+		#region IUIElementOverrides8
+		protected override void OnBringIntoViewRequested(BringIntoViewRequestedEventArgs e)
+		{
+			m_viewportManager.OnBringIntoViewRequested(e);
+		}
+
+		#endregion
+
+		#region IFrameworkElementOverrides
+
+		protected override Size MeasureOverride(Size availableSize)
+		{
+			if (m_isLayoutInProgress)
+			{
+				throw new InvalidOperationException("Reentrancy detected during layout.");
+			}
+
+			if (IsProcessingCollectionChange)
+			{
+				throw new InvalidOperationException("Cannot run layout in the middle of a collection change.");
+			}
+
+			m_viewportManager.OnOwnerMeasuring();
+
+			m_isLayoutInProgress = true;
+			using var layoutInProgress = Uno.Disposables.Disposable.Create(() => m_isLayoutInProgress = false);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'layoutInProgress'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)